node/src/block/builder.rs

```diff
@@ -238,19 +245,24 @@ impl BlockBuilder {
         Ok(())
     }
 
+    fn is_limits_reached(&self) -> bool {
+        self.total_gas_used > 1_000_000
```

Comment:
This limit should be obtained from blockchain config (p20 for masterchain and p21 for workchain)